tfhe/src/shortint/mod.rs

```diff
@@ -85,8 +86,12 @@ pub use server_key::{CheckError, CompressedServerKey, ServerKey};
 /// // generate the client key and the server key:
 /// let (cks, sks) = gen_keys(PARAM_MESSAGE_2_CARRY_2);
 /// ```
-pub fn gen_keys(parameters_set: Parameters) -> (ClientKey, ServerKey) {
-    let cks = ClientKey::new(parameters_set);
+pub fn gen_keys<P>(parameters_set: P) -> (ClientKey, ServerKey)
+where
+    P: TryInto<ShortintParameterSet>,
+    <P as TryInto<ShortintParameterSet>>::Error: std::fmt::Debug,
+{
+    let cks = ClientKey::new(parameters_set.try_into().unwrap());
     let sks = ServerKey::new(&cks);
```

Comment:
So this is where later down the refactoring and feature introduction (or in ServerKey::new) we would match the in parameter set to create the ServeKey that holds a classical pbs key, a multi bit pbs  with or without and associated Wopbs Key ?

Reply:
either that or matching on cks.parameters un ServerKey::new